src/server.rs

```diff
@@ -17,10 +17,28 @@ use tokio_core::reactor::{self, Handle};
 use tokio_proto::BindServer;
 use tokio_service::NewService;
 
+cfg_if! {
+    if #[cfg(feature = "tls")] {
+        use native_tls::TlsAcceptor;
+        use tokio_tls::TlsAcceptorExt;
+        use errors::native_to_io;
+        use stream_type::StreamType;
+    } else {}
+}
+
+enum Acceptor {
+    Tcp,
+    #[cfg(feature = "tls")]
+    Tls(TlsAcceptor),
+}
+
 /// Additional options to configure how the server operates.
-#[derive(Clone, Default)]
+#[cfg_attr(feature = "tls", derive(Default))]
+#[cfg_attr(not(feature = "tls"), derive(Clone, Default))]
```

Comment:
Same comment as above: I'd rather impl the intersection of possible traits for each feature.